contracts/periphery/SdexLpTokenDeployer.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: GPL-3
+
+pragma solidity 0.8.19;
+pragma experimental ABIEncoderV2;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '../proxy/SdexBeaconProxy.sol';
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+contract SdexLpTokenDeployer is Ownable {
+    using Address for address;
+    address public lpTokenBeaconAddress;
```

Comment:
i suggest to make it immutable. we can have different active logics and beacons in the future, then we'll just need to deploy another instance of deployer.

Reply:
updated